src/main/kotlin/uk/gov/justice/digital/hmpps/courtregister/resource/CourtBuildingMaintenanceResource.kt

```diff
@@ -0,0 +1,216 @@
+        content = [
+          Content(
+            mediaType = "application/json",
+            schema = Schema(implementation = BuildingDto::class)
+          )
+        ]
+      ),
+      ApiResponse(
+        responseCode = "400",
+        description = "Invalid request to add a building",
+        content = [
+          Content(
+            mediaType = "application/json",
+            schema = Schema(implementation = ErrorResponse::class)
+          )
+        ]
+      ),
+      ApiResponse(
+        responseCode = "401",
+        description = "Unauthorized to access this endpoint",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))]
+      ),
+      ApiResponse(
+        responseCode = "403",
+        description = "Incorrect permissions to make building insert",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))]
+      )
+    ]
+  )
+  fun insertBuilding(
+    @Schema(description = "Court ID", example = "ACCRYC", required = true)
+    @PathVariable @Size(max = 12, min = 2, message = "Court ID must be between 2 and 12") courtId: String,
+    @RequestBody @Valid updateBuildingDto: UpdateBuildingDto
+  ): BuildingDto {
+    val newBuilding = buildingService.insertBuilding(courtId, updateBuildingDto)
+    snsService.sendEvent(EventType.COURT_REGISTER_BUILDING_INSERT, courtId)
+    auditService.sendAuditEvent(
+      EventType.COURT_REGISTER_BUILDING_INSERT.name,
+      mapOf("courtId" to courtId, "building" to newBuilding)
+    )
+
+    return newBuilding
+  }
+
+  @PreAuthorize("hasRole('ROLE_MAINTAIN_REF_DATA') and hasAuthority('SCOPE_write')")
+  @Operation(
+    summary = "Delete specified building",
+    description = "Deletes building information, role required is MAINTAIN_REF_DATA",
+    security = [SecurityRequirement(name = "MAINTAIN_REF_DATA", scopes = ["write"])],
+    responses = [
+      ApiResponse(
+        responseCode = "200",
+        description = "Building Information Deleted",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = BuildingDto::class))]
+      ),
+      ApiResponse(
+        responseCode = "401",
+        description = "Unauthorized to access this endpoint",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))]
+      ),
+      ApiResponse(
+        responseCode = "403",
+        description = "Incorrect permissions to delete building",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))]
+      ),
+      ApiResponse(
+        responseCode = "404",
+        description = "Building ID not found",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))]
+      )
+    ]
+  )
+  @DeleteMapping("/id/{courtId}/buildings/{buildingId}")
+  fun deleteBuilding(
```

Comment:
We will need to think about this, but let's leave for now. I suspect we need away of saying a court building as closed rather than delete ....

Reply:
agreed